tests/unittests/storage/test_storage.py

```diff
@@ -0,0 +1,363 @@
+    return [_generate(base_trial, 'metadata', 'user', value=u) for u in users]
+
+
+@pytest.mark.parametrize('storage', storage_backends)
+class TestStorage:
+    """Test all storage backend"""
+
+    def test_create_experiment(self, storage):
+        """Test create experiment"""
+        with OrionState(experiments=[], database=storage) as cfg:
+            storage = cfg.storage()
+
+            storage.create_experiment(base_experiment)
+
+            experiments = storage.fetch_experiments({})
+            assert len(experiments) == 1, 'Only one experiment in the database'
+
+            experiment = experiments[0]
+            assert base_experiment == experiment, 'Local experiment and DB should match'
+
+    def test_fetch_experiments(self, storage, name='default_name', user='default_user'):
+        """Test fetch expriments"""
+        with OrionState(experiments=[base_experiment], database=storage) as cfg:
+            storage = cfg.storage()
+
+            experiments = storage.fetch_experiments({'name': name, 'metadata.user': user})
+            assert len(experiments) == len(cfg.experiments)
+
+            experiment = experiments[0]
+            assert experiment['name'] == name, 'name should match query'
+            assert experiment['metadata']['user'] == user, 'user name should match query'
+
+    def test_register_trial(self, storage):
+        """Test register trial"""
+        with OrionState(experiments=[base_experiment], database=storage) as cfg:
+            storage = cfg.storage()
+            trial1 = storage.register_trial(Trial(**base_trial))
+            trial2 = storage.get_trial(trial1)
+
+            assert trial1.to_dict() == trial2.to_dict(), 'Trials should match after insert'
+
+    def test_register_lie(self, storage):
+        """Test register lie"""
+        with OrionState(experiments=[base_experiment], database=storage) as cfg:
+            storage = cfg.storage()
+            storage.register_lie(Trial(**base_trial))
+
+    def test_reserve_trial(self, storage):
+        """Test reserve trial"""
+        with OrionState(
+                experiments=[base_experiment], trials=[base_trial], database=storage) as cfg:
+            storage = cfg.storage()
+            experiment = cfg.get_experiment('default_name', 'default_user', version=None)
+
+            trial = storage.reserve_trial(experiment)
+            assert trial is not None
+            assert trial.status == 'reserved'
+
+    def test_fetch_trials(self, storage):
+        """Test fetch experiment trials"""
+        with OrionState(
+                experiments=[base_experiment], trials=generate_trials(), database=storage) as cfg:
+            storage = cfg.storage()
+            experiment = cfg.get_experiment('default_name', 'default_user', version=None)
+
+            trials = storage.fetch_trials(experiment)
+            assert len(trials) == len(cfg.trials), 'trial count should match'
+
+    def test_get_trial(self, storage):
```

Comment:
Should test both trial and uid. With each of them, both of them and none of them passed.